### PR TITLE
SwiftCLI 3.0

### PR DESCRIPTION
**Pull request by jakeheis**
WIP

### CODE REVIEW

**Comment by rbukovansky** [comment]
@jakeheis WOW!

BTW Any chance of global options, like -h (for help of given command) or -v (for verbose)?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+os:
```
Comment (rbukovansky) [inline .travis.yml]: As SwiftCLI can be built on Linux, I would suggest @kylef's `.travis.yml` file:

```
os:
- linux
- osx
language: generic
sudo: required
dist: trusty
osx_image: xcode8
install:
- eval "$(curl -sL https://gist.githubusercontent.com/kylef/5c0475ff02b7c7671d2a/raw/9f442512a46d7a2af7b850d65a7e9bd31edfb09b/swiftenv-install.sh)"
script:
- swift test
```

**Comment by jakeheis** [comment]
@rbukovansky Thanks for the suggestions! I just implemented an initial version of global options and will be adding them to the README shortly.

When it comes to the Travis build, I initially tried to use @kylef's file, but I was stuck with a weird error I didn't have time to investigate fully: https://travis-ci.org/jakeheis/SwiftCLI/jobs/216556861. If you have any idea how to fix that issue, I'd appreciate the help

**Comment by rbukovansky** [comment]
@jakeheis Yes, I do.

You are missing `.swift-version` file in repository -> this indicates to that cURLed script in `.travis.yml` which Swift to download and install.
See https://github.com/rbukovansky/SwiftCLI/tree/cli_3 which resulted in this https://travis-ci.org/rbukovansky/SwiftCLI/builds/217081387 (there are some issues with tests for Linux).

File: Tests/SwiftCLITests/ParameterFillerTests.swift [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+//  SwiftCLI
+//
+//  Created by Jake Heiser on 8/2/14.
+//  Copyright (c) 2014 jakeheis. All rights reserved.
+//
+
+import XCTest
+@testable import SwiftCLI
+
+class CommandArgumentParserTests: XCTestCase {
```
Comment (rbukovansky) [inline Tests/SwiftCLITests/ParameterFillerTests.swift]: Shouldn't this be `class ParameterFillerTests: XCTestCase`?

File: Tests/SwiftCLITests/ParameterFillerTests.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    
+    static var allTests : [(String, (CommandArgumentParserTests) -> () throws -> Void)] {
```
Comment (rbukovansky) [inline Tests/SwiftCLITests/ParameterFillerTests.swift]: Similar here?

**Comment by rbukovansky** [comment]
@jakeheis OK. Little bit of work and now my tree mentioned earlier has all changes neccessary to get all builds done. Hope it helps.

Proof: https://travis-ci.org/rbukovansky/SwiftCLI/builds/217192629

**Comment by jakeheis** [comment]
@rbukovansky Thank you! If you could open a pull request into the `cli_3` branch then I could merge your commits in. Thanks again!

**Comment by rbukovansky** [comment]
@jakeheis Rebased my changes on top of your `cli_3` branch and created PR.

**Pull request merged**